_skycap/the-basics/formations-stencils-and-snapshots.md

```diff
@@ -0,0 +1,121 @@
+permalink: /:collection/:path
+---
+
+## Overview
+
+Kubernetes provides amazing flexibilty to developers and operators. By consolidating all communication through a single API it encourages developing code to build, configure and maintain infrastructure. This brings great benefits but also some challenges. 
+
+As your application evolves, so do the configuration files that are required to run it on the infrastructure: new components like Redis or Memcached are added and old ones removed, application configurations and secrets change and the number of micro-services keeps growing. This means you need to keep your configuration files not only flexible and managable, but also version controlled.
+
+Storing your configuration files in your source code, next to the rest of the code is one option. But what if your application is not store in a single git repository (mono-repo) and is developed and controlled by different teams? How do you ensure configuration changes that are required by developers of a certain service are in lock-step with the code? 
+
+Also with secrets being stored in Kubernetes configuration files, how do you manage access to secrets to your team?
+
+As code and configuration live side by side, there is a need for managing the growing complexity of configuration files, controlling access to them and reusability of them. Skycap Formations, Stencils and Snapshots are the best way to generate, maintain and reuse Kuberentes configuration files for your applications.
+
+## What is a Formation?
+
+Formations are like "deployment destinations" for your application. Each application can have multiple deployment destinations. These destinations can be different Kubernetes clusters or different namespaces on the same cluster. 
+
+Each Formation is a collection of multiple Stencils. While you can put all of your Kuberentes configuration files into a single Stencil, using multiple Stencils per Formation helps with grouping and managing them as they grow in complexity.
+
+<img src="/assets/skycap/formation_schematic.png" width="600px">
+
+A micro-services based application is made up of one or more services. These services are run and managed by Kubernetes as `Deployments` and / or `Services`. To deploy these to Kubernetes you need to generate Kubernetes configuration files for `Namespace`, `Deployment`, `Service`, `ConfigMap` or `Secrets` and possibly more.
+
+In most applications a repeating set of configuration sections are applied to many different services. For example, many services in your application need to share the same environment variables or secrets (like database access details) or have the same data files mounted at the same location.
+
+## What is a Stencil?
+
+Stencils are version controlled configuration templates for Kubernetes. They are used together with Formations to provide a secure, easy and flexible way to generating Kubernetes configuration files based on your application requirements.
```

Comment:
`flexible way *to* generating` should be `flexible way *of* generating`

Reply:
Fixed, thanks.